apiserver/server_test.go

```diff
@@ -191,7 +192,11 @@ func (s *serverSuite) TestNewServerDoesNotAccessState(c *gc.C) {
 	proxy := testing.NewTCPProxy(c, mongoInfo.Addrs[0])
 	mongoInfo.Addrs = []string{proxy.Addr()}
 
-	st, err := state.Open(s.State.ModelTag(), mongoInfo, mongotest.DialOpts(), nil)
+	dialOpts := mongo.DialOpts{
```

Comment:
The timeout in mongotest is 5 minutes. This test deliberately shuts down the mongodb connection. Because the state package calls mgo.Session.Copy before it does any db operation and that causes mgo to reconnect automatically (subject to the timeout) and closing State synchronously terminates its workers, if the worker happens to be executing a db operation, it can take as long as the timeout before it responds to the termination request, thus blocking the State.Close method for that long - i.e. 5 minutes.